README.md

```diff
@@ -611,3 +615,65 @@ status:
   binding:
     name: kafka-eftg9
 ```
+
+## Role-Based Access Control (RBAC)
+
+Kubernetes clusters often utilize [Role-based access control (RBAC)][rbac] to authorize subjects to perform specific actions on resources. When operating in a cluster with RBAC enabled, the service binding reconciler needs permission to read resources that provisioned a service and write resources that services are projected into. This extension defines a means for third-party CRD authors and cluster operators to expose resources to the service binding reconciler. Cluster operators **MAY** impose additional access controls beyond RBAC.
+
+[rbac]: https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+
+### For Cluster Operators and CRD Authors
+
+Cluster operators and CRD authors **MAY** opt-in resources to service binding by defining a `ClusterRole` with a label matching `service.binding/controller=true`. For Provisioned Service-able resources the `get`, `list`, and `watch` verbs **MUST** be granted. For PodSpec-able resources the `get`, `list`, `watch`, `update`, and `patch` verbs **MUST** be granted.
+
+#### Example Resource
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: awesome-service-bindings
+  labels:
+    service.binding/controller: "true" # matches the aggregation rule selector
+rules:
+# for Provisioned Service-able resources only
+- apiGroups:
+  - awesome.example.com
+  resources:
+  - awesomeservices
+  verbs:
+  - get
+  - list
+  - watch
+# for PodSpec-able resources (also compatible with Provisioned Service-able resources)
```

Comment:
I wonder if we should split this into a section for deployed applications, saying something like "if you add `service.binding/controller: "true"` to your application's PodSpec-able resource, the service binding framework will automatically inject the following `ClusterRole`".  

I am thinking it will be hard for the cluster operator to know all of the apiGroups and resources for the possible PodSpec-able resources (CRs from Operators, Knative resources, regular k8s deployments, etc), so perhaps making it easy for any deployments to opt-in to being bound - and not requiring them to include their own ClusterRole each time?

Reply:
> I wonder if we should split this into a section for deployed applications, saying something like "if you add `service.binding/controller: "true"` to your application's PodSpec-able resource, the service binding framework will automatically inject the following `ClusterRole`".

I'm not sure this is technically possible; perhaps I'm misunderstanding your suggestion. The controller would need RBAC permission to view the resource to see the label that says it should be able to view the resource.

> I am thinking it will be hard for the cluster operator to know all of the apiGroups and resources for the possible PodSpec-able resources (CRs from Operators, Knative resources, regular k8s deployments, etc), so perhaps making it easy for any deployments to opt-in to being bound - and not requiring them to include their own ClusterRole each time?

Strictly speaking it need not be the cluster operator, anyone with the ability to create `ClusterRole`s counts. However, the ability to create new ClusterRole resources is typically tightly restricted on production clusters. Relaxing this role may be viewed by some as a vector for privilege escalation within the cluster. AuthZ is an area where the spec should be conservative in its requirements, allowing individual clusters to do what's right for their particular environment.

The goal for this extension isn't to define a complete authZ model, but to define the minimal contract to facilitate compatibility between spec implementations and third-party components within the bounds of k8s RBAC.

Reply:
I had in mind an entity that already had view access to resources in that namespace and it would just check if the deployed app had opted-in to be updated - but I think you have a good point in terms of restricted environments.